src/effects/RealtimeEffectList.h

```diff
@@ -9,18 +9,34 @@
 #ifndef __AUDACITY_REALTIMEEFFECTLIST_H__
 #define __AUDACITY_REALTIMEEFFECTLIST_H__
 
+#include <any>
 #include <vector>
 
 #include "PluginProvider.h" // for PluginID
 #include "UndoManager.h"
 #include "XMLTagHandler.h"
+#include "Observer.h"
 
 class AudacityProject;
 
 class RealtimeEffectState;
 
 class Track;
 
+struct RealtimeEffectListMessage
+{
+   enum class Type
+   {
+      Insert,///<New effect item was added to the list
+      Remove,///<Effect item was removed from the list
+      Move ///<Item position has changed, new position will be stored as size_t value in extra
+   };
+   Type type;
+   size_t index;
+
+   std::any extra;///< depends on message type
```

Comment:
Maybe `std::variant` is the more proper thing when the alternatives are not open-ended but instead all enumerated here :shrug:  To my mind, `std::any` is more appropriate where there is open-endedness (if, say, RealtimeEffectListMessage was the root of a class hierarchy)

Or simply make it a numerical field that is ignored for the other message types.